scan_to_inventory/static/www/controllers/c_select_quantity.js

```diff
@@ -0,0 +1,64 @@
+angular.module('scan_to_inventory').controller(
+        'SelectQuantityCtrl',
+        ['$scope', '$rootScope', 'jsonRpc', '$state', 'StockInventoryModel', '$translate',
+        function ($scope, $rootScope, jsonRpc, $state, StockInventoryModel, $translate) {
+
+    $scope.data = {
+        'qty': '',
+    };
+
+    $scope.$on(
+            '$stateChangeSuccess',
+            function(event, toState, toParams, fromState, fromParams){
+        if ($state.current.name === 'select_quantity') {
+            // Set Focus
+            angular.element(document.querySelector('#input_quantity'))[0].focus();
+            $scope.data.qty = '';
+            // Get Product Data
+            $scope.product = $rootScope.ProductListByEan13[toParams.ean13];
+        }
+    });
+
+    $scope.submit = function () {
+        if ($scope.data.qty !== null && !isNaN($scope.data.qty)){
```

Comment:
typo in your suggestion I guess. the line below works.

    if ($scope.data.qty)

And check about numeric field is managed by the view. It should be enougth. 

    <input type="number" ng-model="data.qty" />

Thanks for pointing this simplification.